docs/core/examples/echoserv_ssl.py

```diff
@@ -16,7 +17,9 @@ def main(reactor):
     certData = getModule(__name__).filePath.sibling("server.pem").getContent()
     certificate = ssl.PrivateCertificate.loadPEM(certData)
     factory = protocol.Factory.forProtocol(echoserv.Echo)
-    reactor.listenSSL(8000, factory, certificate.options())
+    tcpEndpoint = TCP6ServerEndpoint(reactor, 8000)
+    endpoint = wrapServerTLS(certificate.options(), tcpEndpoint, reactor)
+    endpoint.listen(factory)
     return defer.Deferred()
```

Comment:
Not sure if this can explain better what is going on here

I feel that for new users, there is a lot of "magic" happening here.

```suggestion
    # This will be fired when the port is bound,
    # but we return a deferred that is never fired to keep the
    # reactor loop always active.
    deferred = endpoint.listen(factory)
    deferred.addCallback(lambda _: defer.Deferred())
    return deferred
```

Maybe also rewrite this example to `async def main(reactor)` similar to docs/core/examples/ssl_clientauth_client.py to encourage new users to start with "best practices"

Reply:
This is example code; the prose around it ought to be explaining what is happening. I would really like to keep it at zero comments. Expanding the example with lots of inline comments makes the narrative doc harder to read because you're hit with a wall of code that makes it impossible to even see the prose above and below without scrolling back and forth. Inline end-of-line comments to explain some detail might be okay but a whole extra paragraph is too distracting.

Reply:
I had a look at the relevant prose and I was going to try to add something there, but I felt like I already explained it, so I don't think I'm going to change anything here.

We really need a bigger project on this too, to get some new users in to read through the docs and tell us what is wrong (I suspect they will give us a list with a million items on it, and this one will be down in the high hundreds of thousands ;))